### PR TITLE
Fix filter errors

### PR DESCRIPTION
**Pull request by MariaRoz**
Снимок экрана от 2019-09-23 23-16-19
Снимок экрана от 2019-09-23 23-16-40

### CODE REVIEW

**Comment by MariaRoz** [comment]
> What the role of NavigationEnd?

It's an event triggered when a navigation ends successfully

File: src/app/components/homepage/sidebar/sidebar.component.html [hunk 1]
```diff
@@ -21,9 +21,8 @@
         matTooltip="Music"
         matTooltipPosition="right"
         class="add_circle_outline"
-        [routerLink]="['']"
     >
-        <mat-icon [ngClass]="{'sidebar__button--selected': isActive('audio')}" class="sidebar__button sidebar__button--base" (click)="filterByContent('audio')"
+        <mat-icon [ngClass]="{'sidebar__button--selected': isActive('audio') && selectedIcon}" class="sidebar__button sidebar__button--base" (click)="filterByContent('audio')"
```
Comment (didevgen) [inline src/app/components/homepage/sidebar/sidebar.component.html]: you can move selectedIcon into isActive :)

File: src/app/components/homepage/sidebar/sidebar.component.ts [hunk 2]
```diff
@@ -12,31 +13,40 @@ export class SidebarComponent implements OnInit {
...
+    selectedIcon = false;
 
     constructor(
         public router: Router,
         private getDataService: GetDataService,
     ) {}
 
     ngOnInit() {
-        this.render();
+        this.router.events
```
Comment (didevgen) [inline src/app/components/homepage/sidebar/sidebar.component.ts]: unsubscribe

File: src/app/components/homepage/sidebar/sidebar.component.ts [hunk 3]
```diff
@@ -12,31 +13,40 @@ export class SidebarComponent implements OnInit {
...
-
     filterByContent(type) {
+        this.selectedIcon = true;
         this.activeType = type;
         this.filteredMedia = [];
         this.media
             .filter(content => content.contentType === type)
             .map(result => this.filteredMedia.push(result));
         this.getDataService.filterMedia(this.filteredMedia);
+        if (this.router.url !== '/userPage') {
```
Comment (didevgen) [inline src/app/components/homepage/sidebar/sidebar.component.ts]: extract route to constant

**Pull request merged**